modules/effects/src/main/scala/dev/profunktor/redis4cats/extensions/luaScripting.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018-2021 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats.extensions
+
+import cats.effect.kernel.{ Resource, Sync }
+import cats.syntax.all._
+import cats.{ Applicative, ApplicativeThrow }
+import dev.profunktor.redis4cats.algebra.Scripting
+import dev.profunktor.redis4cats.effects.ScriptOutputType
+import io.lettuce.core.RedisNoScriptException
+
+object luaScripting {
+
+  final case class LuaScript(contents: String, sha: String)
+
+  object LuaScript {
+
+    def make[F[_]: Applicative](redis: Scripting[F, _, _])(contents: String): F[LuaScript] =
+      redis.digest(contents).map(sha => LuaScript(contents, sha))
+
+    /** Helper to load a lua script from resources/lua/{resourceName}. The path to the lua scripts can be configured.
+      * @param redis
+      *   redis commands
+      * @param resourceName
+      *   filename of the lua script
+      * @param pathToScripts
+      *   path to the lua scripts
+      * @return
+      *   [[LuaScript]]
+      */
+    def loadFromResources[F[_]: Sync](redis: Scripting[F, _, _])(
+        resourceName: String,
+        pathToScripts: String = "lua"
+    ): F[LuaScript] =
+      Resource
+        .fromAutoCloseable(
+          Sync[F].delay(
+            scala.io.Source.fromResource(resource = s"$pathToScripts/$resourceName")
+          )
+        )
```

Comment:
Are we able to use fs2's resource reading methods here: https://github.com/typelevel/fs2/blob/9aaabbb7bca60af52b809c95fbea5431e27b842e/io/jvm-native/src/main/scala/fs2/io/iojvmnative.scala#L59-L71?

Reply:
What about adding a load from file too using https://github.com/typelevel/fs2/blob/9aaabbb7bca60af52b809c95fbea5431e27b842e/io/shared/src/main/scala/fs2/io/file/Files.scala

Reply:
@mmienko perhaps its worth adding the dependency . I don't see any harm in that

Reply:
Imo, this use case is simple enough to keep the native Scala api. We're reading the whole file, so wouldn't use any of FS2 stream api. Under the hood, FS2 also uses `blocking` under the hood but with the `java.nio.channels.FileChannel`.